src/main/kotlin/no/digdir/informasjonsforvaltning/fdk_dataservice_harvester/harvester/HarvesterActivity.kt

```diff
@@ -0,0 +1,42 @@
+package no.digdir.informasjonsforvaltning.fdk_dataservice_harvester.harvester
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import no.digdir.informasjonsforvaltning.fdk_dataservice_harvester.adapter.HarvestAdminAdapter
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.util.MultiValueMap
+import java.util.*
+import javax.annotation.PostConstruct
+
+private val LOGGER = LoggerFactory.getLogger(HarvesterActivity::class.java)
+
+@Service
+class HarvesterActivity(
```

Comment:
Kan prøve å forklare hva som skjer her. Testet uansett lokalt ved å legge inn delays i hver launch, etter som jeg kunne se så fungerte det slik jeg så for meg.

Så at det å kjøre på GlobalScope var frarådet, så flyttet alt ut i klassen her med en egen scope. Scope som jeg gir konteksten 'Dispatchers.Default', default hørtes mest korrekt ut å bruke. https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-default.html

Funksjonen 'initiateHarvest' starter to jobber. En under variabelen 'harvest' og en loggejobb som venter til 'harvest' er helt ferdig. Innad i 'harvest' starter en ny jobb pr DataSource, da med jobben 'harvest' som kontekst, det medfører at loggejobben, som venter på 'harves't ,ikke kjører før alle DataSources er ferdig høstet.

Reply:
Ganske kult det!